librz/core/cmd/cmd_search_rop.c

```diff
@@ -4,726 +4,402 @@
-			*result = limit + (src1 - src2);
-		} else {
-			*result = src1 - src2;
-		}
-		return true;
-	}
-	if (!strcmp(op, "*")) {
-		*result = src1 * src2;
-		return true;
-	}
-	if (!strcmp(op, "|")) {
-		*result = src1 | src2;
-		return true;
-	}
-	if (!strcmp(op, "/")) {
-		*result = src1 / src2;
-		return true;
-	}
-	if (!strcmp(op, "%")) {
-		*result = src1 % src2;
-		return true;
-	}
-	if (!strcmp(op, "<<")) {
-		*result = src1 << src2;
-		return true;
-	}
-	if (!strcmp(op, ">>")) {
-		*result = src1 >> src2;
-		return true;
-	}
-	if (!strcmp(op, "&")) {
-		*result = src1 & src2;
-		return true;
+	if (reg_idx == 0) {
+		return NULL;
 	}
-	if (!strcmp(op, "+=")) {
-		*result = old_src1 + src2;
-		return true;
+
+	// Check if the register is correct for the given architecture.
+	if (rz_analysis_is_reg_in_profile(core->analysis, reg)) {
+		return strdup(reg);
 	}
-	if (!strcmp(op, "-=")) {
-		if (src2 > old_src1) {
-			*result = limit + (old_src1 - src2);
-		} else {
-			*result = old_src1 - src2;
-		}
-		return true;
+
+	return NULL;
+}
+
+static bool parse_constant(const char *str, int *idx, unsigned long long *value) {
+	int base = 10;
+	int neg = 0;
+	char num_str[256] = { 0 };
+	int num_idx = 0;
+
+	skip_whitespace(str, idx);
+
+	if (str[*idx] == '-') {
+		neg = 1;
+		(*idx)++;
 	}
-	if (!strcmp(op, "*=")) {
-		*result = old_src1 * src2;
-		return true;
+
+	skip_whitespace(str, idx);
+
+	if (str[*idx] == '0' && (str[*idx + 1] == 'x' || str[*idx + 1] == 'X')) {
+		base = 16;
+		*idx += 2;
 	}
-	if (!strcmp(op, "/=")) {
-		*result = old_src1 / src2;
-		return true;
+
+	while (isdigit(str[*idx]) || (base == 16 && isxdigit(str[*idx]))) {
+		num_str[num_idx++] = str[*idx];
+		(*idx)++;
 	}
-	if (!strcmp(op, "%=")) {
-		*result = old_src1 % src2;
-		return true;
+
+	if (num_idx == 0) {
+		return false;
 	}
-	if (!strcmp(op, "<<")) {
-		*result = src1 << src2;
-		return true;
+
+	*value = strtoull(num_str, NULL, base);
+	if (neg) {
+		*value = -*value;
 	}
-	if (!strcmp(op, ">>")) {
-		*result = src1 >> src2;
-		return true;
+
+	return true;
+}
+
+static bool parse_reg_to_const(RzCore *core, char *str, RzRopConstraint *rop_constraint) {
+	int idx = 0;
+	char *dst_reg = parse_register(core, str, &idx);
+	if (!dst_reg) {
+		return false;
 	}
-	if (!strcmp(op, "&=")) {
-		*result = src1 & src2;
-		return true;
+
+	if (!parse_il_equal(str, &idx)) {
+		free(dst_reg);
+		return false;
 	}
-	if (!strcmp(op, "^=")) {
-		*result = src1 ^ src2;
-		return true;
+
+	unsigned long long const_value;
+	if (!parse_constant(str, &idx, &const_value)) {
+		free(dst_reg);
+		return false;
 	}
-	if (!strcmp(op, "|=")) {
-		*result = src1 | src2;
-		return true;
+
+	if (!parse_eof(str, idx)) {
+		free(dst_reg);
+		return false;
 	}
-	return false;
+
+	rop_constraint->type = MOV_CONST;
+	rop_constraint->args[DST_REG] = dst_reg;
+	rop_constraint->args[SRC_REG] = NULL;
+	char value_str[256];
+	snprintf(value_str, sizeof(value_str), "%llu", const_value);
```

Comment:
```suggestion
	rz_strf(value_str, "%llu", const_value);
```

Reply:
`%llu` is not portable, you need to use `"%" PFMT64u` which gets converted to the correct format.

Reply:
after re-reading the code since you are going to use `strdup` anyway, then use `rz_str_newf("%" PFMT64u, const_value);`. apply this to all the `snprintf(value_str, sizeof(value_str), "%llu", const_value);` which you have defined.